app/models/criteria/association_builder.rb

```diff
@@ -5,9 +5,10 @@ def initialize(relation)
       @relation = relation
     end
 
-    def build(conditions)
+    def build(conditions, &plan_finalization)
       plans = aggregate_plans(conditions)
-      finalize_plans(plans)
+      plan_finalization.call(plans)
```

Comment:
Performance/RedundantBlockCall: Use `yield` instead of `plan_finalization.call`.